.circleci/config.yml

```diff
@@ -0,0 +1,104 @@
+
+version: 2
+jobs:
+  build:
+    docker:
+      - image: openjdk:8
+        environment:
+          CI_TERRIBLENESS: 30.seconds
+        parallelism: 2
+    working_directory: ~/linkerd
+    steps:
+      - checkout
+
+      - setup_remote_docker
+
+      - restore_cache:
+          key: linkerd-{{ checksum "project/LinkerdBuild.scala" }}-{{ checksum "admin/src/main/resources/io/buoyant/admin/package.json" }}
```

Comment:
will add some comments.

re: LinkerdBuild.scala, i tried to pick a file which, when changed, would best motivate invalidating `~/.ivy2` in the circleci cache. if there's a better file, i'm open to it.

related: any idea why we weren't caching `~/.ivy2` in circleci 1.0? it's a huge time saver.